apps/src/util/dropletCommon.js

```diff
@@ -0,0 +1,62 @@
+/*globals dashboard*/
+import i18n from '@cdo/locale';
+import trackEvent from '@cdo/apps/util/trackEvent';
+
+/**
+ * Runs the user's droplet code through the droplet parser. If droplet is unable
+ * to parse the user's code, call the error callback with any information we
+ * have regarding the error. Additionally, this will return `true` if any
+ * errors are detected and `false` otherwise.
+ * @param {object} dropletEditor The droplet engine
+ * @param {function} errorCallback The function to call if an error is detected.
+ *                   Usage: errorCallback(lineNumber, errorMessage)
+ * @returns {boolean} true if an error was detected. false otherwise.
+ */
+export const findDropletParseErrors = (dropletEditor, errorCallback) => {
+  if (!dropletEditor) {
+    return false;
+  }
+
+  try {
+    dropletEditor.parse();
+  } catch (error) {
+    // Example error message: 'Line ###. Error Message'
+    let matchedLineNumber = error.message.match(/Line (\d+)./);
+    if (matchedLineNumber) {
+      // We were able to find the line number. Report it to the user.
+      let errorMessage = error.message.match(/Line \d+. (.*)/)[1];
+      if (error.message.includes('indent must be inside block')) {
```

Comment:
this feels delicate because this would break if we change the error message or make the error translatable. although i don't know that i have any good alternatives to suggest outside of moving this string into a constant that this file and `droplet-full.js` share. thoughts?

Reply:
Curious what you think about doing something like this: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/acemode/errorMapper.js

Reply:
yeah, i like this approach. would it make sense to add this error to `errorMapper.js`, or would we create a new/separate file?

Reply:
Hmm I could go either way. It may be nice to reuse `errorMapper.js` in order to keep this all in one place.

Reply:
yeah, agreed. if it makes sense there, let's just add it to the existing file